kong/db/schema/init.lua

```diff
@@ -1703,6 +1705,28 @@ local function collect_field_reference(refs, key, reference)
 end
 
 
+local function validate_deprecation_exclusiveness(data, shorthand_value, shorthand_name, shorthand_definition)
+  if shorthand_definition.deprecation and shorthand_definition.deprecation.replaced_with then
+    for _, replaced_with_element in ipairs(shorthand_definition.deprecation.replaced_with) do
+      local new_field_value = replaced_with_element.translation and replaced_with_element.translation(data)
+                                                                or table_path(data, replaced_with_element.path)
+
+      if new_field_value and not deepcompare(new_field_value, shorthand_value) then
+        local new_field_name = join_string(".", replaced_with_element.path)
+
+        return nil, string.format(
+          "both deprecated and new field are used but their values mismatch: %s = %s vs %s = %s",
+          shorthand_name, inspect(shorthand_value),
+          new_field_name, inspect(new_field_value)
```

Comment:
If I am right, `inspect()` prints the object across multiple lines. Is this the desired effect? May interfere with log analysing tools?

Reply:
I removed the inspect ✅